packages/gatsby/src/bootstrap/index.js

```diff
@@ -75,6 +75,12 @@ module.exports = async (args: BootstrapArgs) => {
     getConfigFile(program.directory, `gatsby-config`)
   )
 
+  if (typeof config === `function`) {
+    report.panic(
+      `Custom gatsby-config as a Function is not supported in site root.`
+    )
+  }
+
   // theme gatsby configs can be functions or objects
```

Comment:
Should we update this comment as well?

Reply:
This seems fine. What would you want to update it to?

Reply:
We now panic and crash if the `gatsby-config` is a function, this seems to be the opposite of what the comment says. Unless the root config is not considered a theme config in which case I misunderstood